sdk/resourcemanager/azure-resourcemanager-monitor/src/main/java/com/azure/resourcemanager/monitor/implementation/DiagnosticSettingImpl.java

```diff
@@ -219,8 +222,10 @@ public boolean isInCreateMode() {
 
     @Override
     public Mono<DiagnosticSetting> createResourceAsync() {
-        this.innerModel().withLogs(new ArrayList<>(logSet.values()));
-        this.innerModel().withLogs(new ArrayList<>(logCategoryGroupSet.values()));
+        this.innerModel()
+            .withLogs(new ArrayList<>(
+                Stream.concat(logSet.values().stream(),
+                logCategoryGroupSet.values().stream()).distinct().collect(Collectors.toList())));
```

Comment:
nit, distinct() seems unnecessary IMHO. Currently we don't have equals() defined in LogSettings, so it has no effect anyways..

Reply:
I was thinking in case a metric name and log name overlapped (not likely I suppose).  Happy to remove.  I'm away from my computer now but I can update later today.

Reply:
I believe metrics are in `metricsSet`?
https://github.com/Azure/azure-sdk-for-java/blob/6fc1ec62fe97589c512dc7b1193f1ad5472d8d53/sdk/resourcemanager/azure-resourcemanager-monitor/src/main/java/com/azure/resourcemanager/monitor/implementation/DiagnosticSettingImpl.java#L106

It's ok, not hurting to have the deduplication.